pkg/instancegroups/instancegroups.go

```diff
@@ -561,3 +561,21 @@ func (c *RollingUpdateCluster) deleteNode(ctx context.Context, node *corev1.Node
 
 	return nil
 }
+
+func (c *RollingUpdateCluster) RollSingleInstance(ctx context.Context, cloudMember *cloudinstances.CloudInstanceGroupMember, detach bool) error {
+	if detach {
+		if cloudMember.CloudInstanceGroup.InstanceGroup.IsMaster() {
+			return fmt.Errorf("cannot detatch master instances. Run with --detatch=false")
+
+		}
+		err := c.detachInstance(cloudMember)
+		if err != nil {
+			return fmt.Errorf("failed to detach instance: %v", err)
```

Comment:
The most likely cause of this would be that the cloud provider doesn't support surging. It might be better to default `detach` to false for such cloud providers.

Reply:
I missed that this was a param passed on. Changed the functions to figure out the instance group role on its own instead.